### PR TITLE
fix the gradient backward issue when joint training with s3prl frontend

### PR DESCRIPTION
**Pull request by simpleoier**
In s3prl, `feature_grad_mult` was set to 0. Thus the forward is in the context of `torch.no_grad()` [here](https://github.com/s3prl/s3prl/blob/57c3ba0b2215edf0aacea968d16493788a48a424/s3prl/upstream/wavlm/WavLM.py#L361C1-L367). It will stop the gradient back-propagation in joint training. To fix it, just set it to be 1 manually

### CODE REVIEW

**Comment by sw005320** [comment]
@Emrys365, can you review this PR?
I hope this would solve our issues completely.

File: espnet2/asr/frontend/s3prl.py [hunk 1]
```diff
@@ -50,14 +50,10 @@ def __init__(
...
+            if getattr(upstream.upstream.model, "feature_grad_mult", None):
+                upstream.upstream.model.feature_grad_mult = 1.0
         upstream.eval()
-        if getattr(
-            upstream.upstream, "model", None
-        ) is not None and upstream.upstream.model.__class__.__name__ in [
-            "Wav2Vec2Model",
-            "HubertModel",
-        ]:
-            upstream.upstream.model.encoder.layerdrop = 0.0
```
Comment (Emrys365) [inline espnet2/asr/frontend/s3prl.py:60]: Why are these lines removed?

**Review by Emrys365 — APPROVED** [review]
LGTM. I can also verify the gradient issue is resolved on my side.

**Review by Emrys365 — APPROVED** [review]
LGTM. I can also verify the gradient issue is resolved on my side.

File: espnet2/asr/frontend/s3prl.py (hunk 1 above)
Comment (pengchengguo) [reply espnet2/asr/frontend/s3prl.py:60]: I think this is because S3PRL already sets `encoder_layerdrop=0` when initializing an upstream, so we no longer need to do this in ESPnet. Am I right?

Comment (simpleoier) [reply espnet2/asr/frontend/s3prl.py:60]: Yes, it is correct.

**Comment by sw005320** [comment]
Thanks a lot for fixing it!
